File: backend/models/device.go
package models

import (
	"time"
)

type Device struct {
	ID           uint              `gorm:"primaryKey;autoIncrement" json:"id"`
	Name         string            `gorm:"type:varchar(128);not null;index" json:"name"`
	DeviceType   string            `gorm:"type:varchar(64);not null" json:"device_type"`
	Vendor       string            `gorm:"type:varchar(32)" json:"vendor"`
	Model        string            `gorm:"type:varchar(64)" json:"model"`
	Role         string            `gorm:"type:varchar(32);default:access" json:"role"`
	ManagementIP string            `gorm:"type:varchar(45);not null;uniqueIndex" json:"management_ip"`
	SSHPort      int               `gorm:"default:22" json:"ssh_port"`
	CredentialID *uint             `gorm:"index" json:"credential_id"`
	Credential   *Credential       `gorm:"foreignKey:CredentialID" json:"credential,omitempty"`
	GroupID      *uint             `gorm:"index" json:"group_id"`
	Group        *DeviceGroup      `gorm:"foreignKey:GroupID" json:"group,omitempty"`
	Building     string            `gorm:"type:varchar(64)" json:"building"`
	Floor        int               `gorm:"default:0" json:"floor"`
	APName       string            `gorm:"type:varchar(128)" json:"ap_name"`
	SNMPCommunity string           `gorm:"type:varchar(64)" json:"snmp_community"`
	Status       int               `gorm:"default:0;index" json:"status"`
	LastSeen     *time.Time        `json:"last_seen"`
	Description  string            `gorm:"type:varchar(256)" json:"description"`
	CreatedAt    time.Time         `json:"created_at"`
	UpdatedAt    time.Time         `json:"updated_at"`
}

func (Device) TableName() string {
	return "devices"
}

File: frontend/src/api/device.ts
import request from './request'
import type { PageData } from './request'

export interface Device {
  id: number
  name: string
  device_type: string
  vendor: string
  model: string
  role: string
  management_ip: string
  ssh_port: number
  credential_id: number | null
  group_id: number | null
  building: string
  floor: number
  ap_name: string
  status: number
  last_seen: string | null
  description: string
  created_at: string
  updated_at: string
  group?: { id: number; name: string }
  credential?: { id: number; name: string }
}

export interface DeviceStats {
  total: number
  online: number
  offline: number
  role_counts: Record<string, number>
}

export function getDevices(params?: Record<string, any>) {
  return request.get<PageData<Device>>('/devices', { params })
}

export function getAllDevices() {
  return request.get<Device[]>('/devices/all')
}

export function getDevice(id: number) {
  return request.get<Device>(`/devices/${id}`)
}

export function createDevice(data: Partial<Device>) {
  return request.post<Device>('/devices', data)
}

export function updateDevice(id: number, data: Partial<Device>) {
  return request.put(`/devices/${id}`, data)
}

export function deleteDevice(id: number) {
  return request.delete(`/devices/${id}`)
}

export function getDeviceStats() {
  return request.get<DeviceStats>('/devices/stats')
}

export function inspectDevice(id: number) {
  return request.post(`/devices/${id}/inspect`)
}

export function batchInspect(deviceIds: number[]) {
  return request.post('/inspections/batch', { device_ids: deviceIds })
}

export function getInspectionReport(params?: Record<string, any>) {
  return request.get('/inspections/report', { params })
}

export function getLatestInspections(limit = 10) {
  return request.get('/inspections/latest', { params: { limit } })
}

export function getDashboard() {
  return request.get('/monitor/dashboard')
}

export function getAlerts(params?: Record<string, any>) {
  return request.get('/alerts', { params })
}

export function getAlertStats() {
  return request.get('/alerts/stats')
}

export function confirmAlert(id: number) {
  return request.put(`/alerts/${id}/confirm`)
}

export function resolveAlert(id: number) {
  return request.put(`/alerts/${id}/resolve`)
}
